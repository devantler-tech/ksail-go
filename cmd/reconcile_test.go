package cmd_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
)

func TestNewReconcileCmd(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandCreation(t, cmd.SimpleCommandTestData{
		CommandName:   "reconcile",
		NewCommand:    cmd.NewReconcileCmd,
		ExpectedUse:   "reconcile",
		ExpectedShort: "Reconcile workloads in the cluster",
	})
}

func TestReconcileCmd_Execute(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandExecution(t, cmd.SimpleCommandTestData{
		CommandName: "reconcile",
		NewCommand:  cmd.NewReconcileCmd,
	})
}

func TestReconcileCmd_Help(t *testing.T) {
	t.Parallel()

	cmd.TestSimpleCommandHelp(t, cmd.SimpleCommandTestData{
		CommandName: "reconcile",
		NewCommand:  cmd.NewReconcileCmd,
	})
}
