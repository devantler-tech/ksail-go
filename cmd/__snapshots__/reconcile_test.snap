[TestReconcileCmd_Help - 1]
Reconcile workloads in the cluster to match the desired state
defined in configuration files.

Usage:
  reconcile [flags]

Flags:
  -c, --context string                           Kubernetes context to reconcile workloads in (default "kind-ksail-default")
  -h, --help                                     help for reconcile
  -k, --kubeconfig string                        Path to kubeconfig file (default "~/.kube/config")
  -r, --reconciliation-tool ReconciliationTool   Tool to use for reconciling workloads (default Kubectl)
  -s, --source-directory string                  Directory containing workloads to reconcile (default "k8s")

---

[TestReconcileCmd_Execute - 1]
✔ Workloads reconciled successfully (stub implementation)
► Reconciliation tool: Kubectl
► Source directory: k8s
► Context: kind-ksail-default

---